### PR TITLE
(feature) optional param added to throwIfNotFound method

### LINKED ISSUE #1722: [Feature Request] Optional message param to throwIfNotFound
**Describe the solution you'd like**
Would love to be able to provide a message string to the throwIfNotFound method.

**Describe alternatives you've considered**
N/A

**Additional context**
Would help solve the pain of writing a few lines of code to throw a custom error every time a necessary object doesn't exist.

### PR DESCRIPTION
**Pull request by vasdhara**
closes #1722 

Is this what was required for the feature request? Please let me know if I have missed anything. A first-time contributor so slightly overwhelmed.

### CODE REVIEW

File: doc/api/query-builder/other-methods.md [hunk 1]
```diff
@@ -1071,13 +1071,19 @@ console.log(firstPerson.age);
...
 
 Causes a [Model.NotFoundError](/api/types/#class-notfounderror) to be thrown if the query result is empty.
 
 You can replace `Model.NotFoundError` with your own error by implementing the static [Model.createNotFoundError(ctx)](/api/model/static-methods.html#static-createnotfounderror) method.
 
+##### Arguments
+
+| Argument | Type   | Description                      |
+| -------- | ------ | -------------------------------- |
+| data     | object | optional object with custom data |
```
Comment (koskimas) [inline doc/api/query-builder/other-methods.md:1085]: There's no way for people to use this based on this documentation. Where does this data go? What properties can it have? How are different properties treated (like the special case for `message`).

File: tests/integration/find.js [hunk 2]
```diff
@@ -574,7 +574,7 @@ module.exports = session => {
             });
         });
 
-        it('.throwIfNotFound() with empty result', done => {
+        it('custom .throwIfNotFound() with empty result', done => {
```
Comment (koskimas) [inline tests/integration/find.js:577]: Why did you add the word `custom` here?

**Comment by koskimas** [comment]
Hi @vasdhara! Thank you so much for your contribution! I added a couple of comments. You also need to update the typescript typings.

File: tests/integration/find.js (hunk 2 above)
Comment (vasdhara) [reply tests/integration/find.js]: was a typo, reverted

File: doc/api/query-builder/other-methods.md (hunk 1 above)
Comment (vasdhara) [reply doc/api/query-builder/other-methods.md]: I've updated the description and added a custom type example, is that sufficient?

**Comment by vasdhara** [comment]
@koskimas thanks. added fixes. Also, added a new interface for throwIfNotFound typings, since I couldn't find a previously written interface that could be reused in this case.

File: doc/api/query-builder/other-methods.md (hunk 1 above)
Comment (koskimas) [reply doc/api/query-builder/other-methods.md]: That's excellent!
